src/Util/Util.hs

```diff
@@ -56,7 +56,7 @@ listDirectoryRecursive path = do
   x >>= return . (>>= id)
 
 zipMap :: Ord k => Map.Map k a -> Map.Map k b -> Map.Map k (a, b)
-zipMap m1 m2 = Map.mapWithKey (\k v1 -> (v1, m2 Map.! k)) m1
+zipMap = Map.intersectionWith (,)
```

Comment:
I recommend using [Hoogle](https://hoogle.haskell.org/) to find out if there are any functions with the specific signature you're looking for. Since in Haskell the function name and signature often reveal intent and behavior, we can get to this one through some intelligent querying:

```bash
$ hoogle 'Map k a -> Map k b -> Map k (a,b)'
Streamly.Internal.Data.Stream.IsStream.Top crossJoin :: Monad (t m) => t m a -> t m b -> t m (a, b)
```

That's not exactly what we want, or at least not immediately obvious that it might be so. Instead, let's see if we can apply a function to both values to create a third one, mapping to it:

```bash
$ hoogle '(a -> b -> c) -> Map k a -> Map k b -> Map k c'
Data.Map.Internal intersectionWith :: Ord k => (a -> b -> c) -> Map k a -> Map k b -> Map k c
Data.Map.Lazy intersectionWith :: Ord k => (a -> b -> c) -> Map k a -> Map k b -> Map k c
Data.Map.Strict intersectionWith :: Ord k => (a -> b -> c) -> Map k a -> Map k b -> Map k c
Data.Map.Strict.Internal intersectionWith :: Ord k => (a -> b -> c) -> Map k a -> Map k b -> Map k c
RIO.Map intersectionWith :: Ord k => (a -> b -> c) -> Map k a -> Map k b -> Map k c
Dhall.Map intersectionWith :: Ord k => (a -> b -> c) -> Map k a -> Map k b -> Map k c
Data.HashMap intersectionWith :: Ord k => (a -> b -> c) -> Map k a -> Map k b -> Map k c
Streamly.Internal.Data.Stream.IsStream.Common zipWith :: (IsStream t, Monad m) => (a -> b -> c) -> t m a -> t m b -> t m c
Streamly.Internal.Data.Stream.IsStream.Expand zipWith :: (IsStream t, Monad m) => (a -> b -> c) -> t m a -> t m b -> t m c
Streamly.Prelude zipWith :: (IsStream t, Monad m) => (a -> b -> c) -> t m a -> t m b -> t m c
```

**Aha!** `intersectionWith` sounds like it might do the trick. Then we read the documentation, and it turns out it's what we cant.